T/Thrift/bundled/patches/freebsd.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/lib/cpp/src/thrift/transport/TSocket.cpp b/lib/cpp/src/thrift/transport/TSocket.cpp
+index 9991fd6bd..e46daeba8 100644
+--- a/lib/cpp/src/thrift/transport/TSocket.cpp
++++ b/lib/cpp/src/thrift/transport/TSocket.cpp
+@@ -432,6 +432,10 @@ void TSocket::local_open() {
+   TWinsockSingleton::create();
+ #endif // _WIN32
+
++#if __FreeBSD__
++  #define EAI_NODATA   EAI_NONAME
++#endif // __FreeBSD__
++
```

Comment:
I'm not very convinced by this patch.  How did you come up with that?  `EAI_NODATA` is in `netdb.h` on FreeBSD with value 7, `EAI_NONAME` has value 8 (but `EAI_NODATA` is inside an `#if 0`):
```console
% julia --compile=min -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("x86_64", "freebsd"))'
[ Info: Running privileged container via `sudo`, may ask for your password:
sandbox:${WORKSPACE} # grep -B2 -A2 -r '\<EAI_NODATA\>' /opt/${target}/${target}/sys-root/include/
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/netdb.h-#if 0
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/netdb.h-/* Obsoleted on RFC 2553bis-02 */
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/netdb.h:#define     EAI_NODATA       7      /* no address associated with hostname */
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/netdb.h-#endif
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/netdb.h-#define     EAI_NONAME       8      /* name does not resolve */
--
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/roken-common.h-#define EAI_FAMILY   5       /* ai_family not supported */
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/roken-common.h-#define EAI_MEMORY   6       /* memory allocation failure */
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/roken-common.h:#define EAI_NODATA   7       /* no address associated with nodename */
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/roken-common.h-#define EAI_NONAME   8       /* nodename nor servname provided, or not known */
/opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/roken-common.h-#define EAI_SERVICE  9       /* servname not supported for ai_socktype */
```

Reply:
I'm not a C++ programmer so I could be wrong here. I got the following error on freebsd

```
[12:12:24] /workspace/srcdir/thrift/lib/cpp/src/thrift/transport/TSocket.cpp:461:16: error: use of undeclared identifier 'EAI_NODATA'
[12:12:24]       error == EAI_NODATA
[12:12:24]                ^
[12:12:24] 1 error generated.
```

in a [previous build](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=21922&view=logs&j=a1cbbe6a-f7ab-5046-169a-68359cb1f688&t=b53c2c9f-b070-50cd-cd8b-bccd490e40db&l=417). 

There are many hits when you google this, but essentially FreeBSD obsoleted it in an RFC, as it states in your `netdb.h` example. Similarly, the manual doesn't mention it anymore: https://www.freebsd.org/cgi/man.cgi?query=gai_strerror&sektion=3&apropos=0&manpath=FreeBSD+7.1-RELEASE and the [mailing list](https://lists.freebsd.org/pipermail/freebsd-ports/2003-October/005757.html) states that: 

> Now, getaddrinfo(3) returns EAI_NONAME instead of
EAI_NODATA.

Eventlet patched this in a [similar way](https://github.com/eventlet/eventlet/commit/4e775e23f312c386929e0c5e9a9c952c1ec80c45).

Reply:
Ok, if FreeBSD devs tell to use `EAI_NONAME` instead of `EAI_NODATA` then this is good I guess.